common4j/src/main/com/microsoft/identity/common/java/net/UrlConnectionHttpClient.java

```diff
@@ -324,8 +319,15 @@ private HttpURLConnection setupConnection(HttpRequest request) throws IOExceptio
             urlConnection.setRequestProperty(entry.getKey(), entry.getValue());
         }
 
-        urlConnection.setRequestMethod(request.getRequestMethod());
+        if (urlConnection instanceof HttpsURLConnection) {
```

Comment:
Should we check based on the protocol used in the URL?

Reply:
Yes. This is exactly what it is. URL.openConnection() returns HttpsURLConnection for HTTPS requests, and HttpUrlConnection for HTTP request.

Reply:
added a check and 2 test cases.